src/components/tree/infoPanels/MutationTable.js

```diff
@@ -59,24 +65,79 @@ const TableFirstColumn = styled.td`
   white-space: nowrap;
   vertical-align: baseline;
 `;
-const ListOfMutations = ({name, muts, displayAsIntervals, isNuc}) => {
-  let mutString, title;
+
+/**
+ * Returns a clickable text-like element which (when clicked) changes the
+ * color-by to show the chosen mutation. Shift-clicking will add the position to
+ * the color-by (where possible). Holding command/windows will modify the filters
+ * rather than the color-by.
+ *
+ * (Now that these are clickable it would be fun to display them as badges, similar
+ * to Nextclade)
+ */
+const UnconnectedSingleMutation = ({gene, mutation, currentColorBy, gtFilters, genomeMap, dispatch}) => {
+  function onClick(event) {
+    const {pos, to} = parseMutation(mutation);
+    if (event.metaKey) {
+      // cmd key (mac), windows key behaviour: filter to that mutated state
+      // (if shift is also pressed then we add-to / remove-from the filters)
+      const value = `${gene} ${pos}${to}`; // the requested filter value
+      if (event.shiftKey) {
+        const alreadyActive = (new Set(gtFilters.filter((f) => f.active).map((f) => f.value))).has(value)
+        dispatch(applyFilter(alreadyActive ? 'inactivate' : 'add', genotypeSymbol, [value]))
+      } else {
+        dispatch(applyFilter('focus', genotypeSymbol, [value]))
+      }
+    } else if (event.shiftKey) {
+      // Shift-click behaviour (without command key): add to existing color-by (where possible)
+      const colorBy = encodeColorByGenotypeCumulative({gene, position: pos, currentColorBy, genomeMap});
+      dispatch(changeColorBy(colorBy));
+    } else {
+      // Normal click behaviour: replace previous color-by
+      const colorBy = encodeColorByGenotype({gene, positions: [pos]});
+      dispatch(changeColorBy(colorBy));
+    }
+  }
+  
+  return (<Button onClick={onClick}>
+    {mutation}
+  </Button>)
+}
+
+const SingleMutation = connect((state) => ({
+  genomeMap: state.entropy.genomeMap,
+  currentColorBy: state.controls.colorBy,
+  gtFilters: state.controls.filters[genotypeSymbol] || [],
+}))(UnconnectedSingleMutation);
```

Comment:
Took me a second to remember how `connect` works/wrap my head around this 😅 I pushed up a [commit that uses hooks instead of `connect`](https://github.com/nextstrain/auspice/commit/62a6c1fed1a794347bf765c5ffe30b88359b4c57) in case you're interested.

Reply:
Ha! I'm in the opposite camp, I have never found hooks to be liberating cf. lifecycle methods (beyond little toy examples). We make extensive use of both styles in Auspice and I hope React never forces us to conform to one.